src/librustc_driver/driver.rs

```diff
@@ -379,6 +381,9 @@ impl<'a, 'b, 'ast, 'tcx> CompileState<'a, 'b, 'ast, 'tcx> {
                          cstore: &'a CStore)
                          -> CompileState<'a, 'b, 'ast, 'tcx> {
         CompileState {
+            //Registry initialization should go before the krate initialization
+            //because krate: Some(krate) needs move.
```

Comment:
I don't think this comment is needed, though, -- if someone tries to move `krate: Some(krate)` above `registry: ...` for some reason, they'll get the same error you got.